redis/asyncio/connection.py

```diff
@@ -763,7 +763,11 @@ async def send_packed_command(
             raise ConnectionError(
                 f"Error {err_no} while writing to socket. {errmsg}."
             ) from e
-        except Exception:
+        except BaseException:
+            # The send_packed_command api does not support re-trying a partially
+            # sent message, so there is no point in keeping the connection open.
+            # An unknown number of bytes has been sent and the connection is therefore
```

Comment:
Even if you knew how many bytes were sent (I think you can know), the problem is you can't "take them back".

Reply:
It's not about "taking them back", its about "trying again to send the rest".  This could (theoretically) be done in case of a timeout, and is the way that many `socket.sendall()` implementations work.  (the underlying `socket.send()` sends a potential subset of the data and needs to be retried.)

send_packed_api() could be architected in a similar way, but maintain an internal buffer of unsent data so that it can be re-tried.

Reply:
But of course it's not reasonable to flush a command after an arbitrary delay (could be milliseconds, could be days). The "workaround" supported by RESP is not reasonable for what Redis is.

Reply:
I'm not sure I understand what you mean.   All I'm saying is that sometimes your `send()` can time out, for various reasons, buffers, server responsiveness, or whatever.   It's technically possible to engineer things such that you can at least be sure that you've delivered all of your payload down to the OS to deliver over the wire.    What happens after that is anyone's guess.  Typically, then, you'd listen for the response, and time out in the listening phase if nothing happens.

This is in no way dissimilar to http, and doesn't have anything to do with RESP as such.  

Anyway, I thought I'd comment on that, a _different_ API, allowing one to retry a partially completed `send()` operation, could mean that it would make sense to keep the connection open.  But the current API has no semantics that woudl allow a "retry".

Reply:
also, see the above discussion :)